Jenkinsfile

```diff
@@ -0,0 +1,418 @@
+#!groovy
+
+// Use 'ci-jenkins@somebranch' to pull shared lib from a different branch than the default.
+// Default is configured in Jenkins and should be from "stable" tag.
+// TODO: Remove the branch name here once lib support is merged and tagged
+@Library("ci-jenkins@master") import com.swiftnav.ci.*
+
+String dockerFile = "scripts/Dockerfile.jenkins"
+String dockerMountArgs = "-v /mnt/efs/refrepo:/mnt/efs/refrepo -v /mnt/efs/buildroot:/mnt/efs/buildroot"
+
+String dockerSecArgs = "--cap-add=SYS_PTRACE --security-opt seccomp=unconfined"
+def context = new Context(context: this)
+def builder = context.getBuilder()
+
+pipeline {
+    // Override agent in each stage to make sure we don't share containers among stages.
+    agent any
+    options {
+        // Make sure job aborts eventually if hanging.
+        timeout(time: 8, unit: 'HOURS')
+        timestamps()
+        // Keep builds for 7 days.
+        buildDiscarder(logRotator(daysToKeepStr: '7'))
+    }
+
+    parameters {
+        string(name: "REF", defaultValue: "master", description: "git ref for piksi_buildroot repo")
+        choice(name: "LOG_LEVEL", choices: ['INFO', 'DEBUG', 'WARNING', 'ERROR'])
+        booleanParam(name: "UPLOAD_TO_S3", defaultValue: true, description: "Enable if artifacts should be uploaded to S3")
+    }
+
+    environment {
+        PIKSI_NON_INTERACTIVE_BUILD="y"
+        // Cache package downloads here
+        BR2_DL_DIR="/mnt/efs/buildroot/downloads"
+        // Needed to add multiple private keys
+        SSH_AUTH_SOCK="/tmp/ssh_auth_sock"
+    }
+
+    stages {
+        stage('Build') {
```

Comment:
Do we need a stage to build the base docker image?  Or were you planning on handling that differently (or just leaving it in Travis for now)?

Reply:
Good point, that should be added; but we can rely on the travis build for it for now.
I also need to add that for updating docker images for starling build and piksi-release builds.

Reply:
Sounds good, file an EO ticket?